### DISCUSSION TITLE
Analytics update to v4.7.1 put my install of Nextcloud-AIO v25.0.3 in maintenance mode

### DISCUSSION THREAD

**Discussion by Bruceumba** [nextcloud/all-in-one#1975] [Questions] [answered]
I updated Analytics today (to version 4.7.1), and it broke something.  Went into maintenance mode. I forced it out with an occ command, then it said update needed. Got it back working with: 
`sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ app:disable analytics`
We had a problem like this last month. App writers aren't checking AIO installs, maybe?
Bruce

**Comment by Rello** [comment]
Hi,
no issues on my (non AIO) instances. any log?

**Comment by szaimen** [comment]
Hi @Bruceumba can you run this command and post its output here?

`sudo docker exec --user www-data -it nextcloud-aio-nextcloud find ./ -name analytics -type d -maxdepth 2`

**Comment by Rello** [comment]
in a parallel issue I got this error
`ParseError: syntax error, unexpected '|', expecting ';' or '{' in /var/www/html/nextcloud/apps/analytics/lib/Datasource/Github.php:119`

its related to this line
https://github.com/Rello/analytics/blob/master/lib/Datasource/Github.php#L119

return type declarations should not be the issue - unless AIO is still using PHP7

**Reply by szaimen** [reply]
No, AIO uses php 8.0 currently.

**Reply by szaimen** [reply]
Seems like the user that reported this does not use AIO

**Comment by Rello** [comment]
problem is caused by mixed return types in 2 places, which were introduced in PHP8
https://www.php.net/manual/en/language.types.declarations.php#language.types.declarations.mixed

I will push another php7.4 compatible version and update the min-version in the info.xml with the next release

**Reply by szaimen** [reply]
AIO has php 8.0 so I suspect a different issue here.

**Reply by Rello** [reply]
https://github.com/Rello/analytics/issues/333#issuecomment-1427694994
not sure if this is AIO at all

**Reply by szaimen** [reply]
yes, this is not AIO.

**Reply by Rello** [reply]
thank you for your support

**Reply by szaimen** [reply]
no problem!

**Comment by Rello** [comment, marked as answer]
v4.7.2 was released
